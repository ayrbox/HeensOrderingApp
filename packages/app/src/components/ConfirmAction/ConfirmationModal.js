import React from 'react';
import PropTypes from 'prop-types';

import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import Typography from '@material-ui/core/Typography';

const ConfirmationModal = ({
  open,
  text,
  action,
  onCancel,
}) => (
  <Dialog
    open={open}
    aria-labelledby={text}
    data-testid="confirm-modal"
  >
    <DialogContent>
      <Typography variant="body1" data-testid="message-label">
        {text}
      </Typography>
    </DialogContent>
    <DialogActions>
      <Button
        onClick={onCancel}
        data-testid="no-button"
      >
         No
      </Button>
      <Button
        onClick={action}
        data-testid="yes-button"
      >
        Yes
      </Button>
    </DialogActions>
  </Dialog>
);

ConfirmationModal.propTypes = {
  open: PropTypes.bool.isRequired,
  text: PropTypes.string.isRequired,
  action: PropTypes.func.isRequired,
  onCancel: PropTypes.func.isRequired,
};

export default ConfirmationModal;
