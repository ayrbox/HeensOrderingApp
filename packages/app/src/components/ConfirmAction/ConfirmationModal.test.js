import React from 'react';
import {
  render,
  fireEvent,
  cleanup,
  getNodeText,
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import ConfirmationModal from './ConfirmationModal';

describe('<ConfirmationModal />', () => {
  let container;
  const mockAction = jest.fn();
  const mockCancel = jest.fn();
  const modalText = 'Sample modal text';

  beforeEach(() => {
    container = render(
      <ConfirmationModal
        open
        action={mockAction}
        onCancel={mockCancel}
        text={modalText}
      />,
    );
  });

  afterEach(cleanup);

  it('renders correctly', () => {
    expect(container).toMatchSnapshot();
  });

  it('contains all components', () => {
    const { getByTestId } = container;
    getByTestId(/confirm-modal/);
    getByTestId(/no-button/);
    getByTestId(/yes-button/);
    const message = getByTestId(/message-label/);
    expect(getNodeText(message)).toEqual(modalText);
  });

  it('clicks on yes button', () => {
    const { getByTestId } = container;
    fireEvent.click(
      getByTestId(/no-button/),
    );
    expect(mockCancel).toHaveBeenCalled();
  });

  it('click on no button', () => {
    const { getByTestId } = container;
    fireEvent.click(
      getByTestId(/yes-button/),
    );
    expect(mockAction).toHaveBeenCalled();
  });
});
