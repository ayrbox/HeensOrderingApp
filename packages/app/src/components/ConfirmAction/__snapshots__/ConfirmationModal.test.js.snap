// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmationModal /> renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="overflow: hidden; padding-right: 0px;"
  >
    <div
      aria-hidden="true"
    />
    <div
      aria-labelledby="Sample modal text"
      class="MuiModal-root-15 MuiDialog-root-1"
      data-testid="confirm-modal"
      role="dialog"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root-17"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        class="MuiDialog-container-4 MuiDialog-scrollPaper-2"
        role="document"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        tabindex="-1"
      >
        <div
          class="MuiPaper-root-19 MuiPaper-elevation24-45 MuiPaper-rounded-20 MuiDialog-paper-5 MuiDialog-paperScrollPaper-6 MuiDialog-paperWidthSm-9"
        >
          <div
            class="MuiDialogContent-root-46"
          >
            <p
              class="MuiTypography-root-47 MuiTypography-body1-56"
              data-testid="message-label"
            >
              Sample modal text
            </p>
          </div>
          <div
            class="MuiDialogActions-root-83"
          >
            <button
              class="MuiButtonBase-root-111 MuiButton-root-85 MuiButton-text-87 MuiButton-flat-90 MuiDialogActions-action-84"
              data-testid="no-button"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label-86"
              >
                No
              </span>
              <span
                class="MuiTouchRipple-root-114"
              />
            </button>
            <button
              class="MuiButtonBase-root-111 MuiButton-root-85 MuiButton-text-87 MuiButton-flat-90 MuiDialogActions-action-84"
              data-testid="yes-button"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label-86"
              >
                Yes
              </span>
              <span
                class="MuiTouchRipple-root-114"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
